File: admin/backend/modules/cms/models/testimonial.go
package models

import (
	"aldev/modules/global/models"
)

type Testimonial struct {
	models.BaseModel
	Name    *string `json:"name" gorm:"type:varchar(300)"`
	Role    *string `json:"role" gorm:"type:varchar(300)"`
	Image   *string `json:"image,omitempty" gorm:"type:text;omitempty"`
	Quote   *string `json:"quote" gorm:"type:text"`
	Galleries *string `json:"galleries,omitempty" gorm:"type:text;omitempty"`
}

File: src/types/testimonial.ts
export type TestimonialItem = {
   id: string;
   name: string;
   role: string;
   image?: string;
   quote: string;
   galleries?: string | File[];
   created_at: string;
   updated_at: string;
};

export type TestimonialSchema = {
   name: string;
   role: string;
   image?: File | string;
   quote: string;
   galleries?: (File | string)[];
};

export type ITestimonialTableFilters = {
   name: string;
};
